csaf_2.0/prose/csaf-v2-editor-draft.md

```diff
@@ -3352,6 +3350,125 @@ This does not exclude that such a document is also available in an access protec
 
 > Reasoning: If an advisory is already in the media, an end user should not be forced to collect the pieces of information from a press release but be able to retrieve the CSAF document.
 
+### 5.1.5 Requirement 5: security.txt
+
+In the security.txt there MUST be at least one field `CSAF` which points to either the ROLIE service document or a directory with CSAF files. If this field indicates a web URI, then it MUST begin with "https://" (as per section 2.7.2 of [RFC7230]).
+
+> At the time of this writing, the security.txt is still a proposed standard. The `CSAF` field has not been officially added yet.
+
+Examples:
+
+```
+CSAF: https://domain.tld/security/data/csaf/
+CSAF: https://psirt.domain.tld/advisories/csaf/
+CSAF: https://domain.tld/security/csaf/csaf-service.json
+```
+
+### 5.1.6 Requirement 6: DNS path
+
+The DNS record `csaf.data.security.domain.tld` SHALL resolve as a webserver which either serves directly the ROLIE service document or a directory with CSAF files.
+
+### 5.1.7 Requirement 7: One folder per year
+
+The CSAF documents must be located within folders named `<YYYY>` where `<YYYY>` is the year given in the value of `/document/tracking/initial_release_date`.
+
+Examples:
+
+```
+2021
+2020
+```
+
+### 5.1.8 Requirement 8: index.txt
+
+The index.txt file within MUST provide a list of all filenames of CSAF documents which are located in the sub-directories with their filenames.
+
+Examples:
+
+```
+2020/example_company_-_2020-yh4711.json
+2019/example_company_-_2019-yh3234.json
+2018/example_company_-_2018-yh2312.json
+```
+
+> This can be used to download all CSAF documents.
+
+### 5.1.9 Requirement 9: changes.csv
+
+The file changes.csv must contain the filename as well as the value of `/document/tracking/current_release_date` for each CSAF document in the sub-directories without a heading; lines must be sorted by the latter one.
+
+Examples:
+
+```
+2020/example_company_-_2020-yh4711.json, "2020-07-01T10:09:07Z"
+2018/example_company_-_2018-yh2312.json, "2020-07-01T10:09:01Z"
+2019/example_company_-_2019-yh3234.json, "2019-04-17T15:08:41Z"
+2018/example_company_-_2018-yh2312.json, "2019-03-01T06:01:00Z"
+```
+
+### 5.1.10 Requirement 10: Directory listings
+
+Directory listing SHALL be enabled to support manual navigation.
+
+### 5.1.11 Requirement 11: ROLIE service document
+
+The ROLIE service document MUST be a JSON file that conforms with [RFC8322].
+
+**TODO: Provide Example**
```

Comment:
```suggestion
### 5.1.11 Requirement 11: ROLIE feed document

Resource-Oriented Lightweight Information Exchange (ROLIE) is a proposed standard to ease discovery of security content. ROLIE is built on top of the Atom Publishing Format and Protocol, with specific requirements that support publishing security content. All CSAF documents MUST be listed in a single ROLIE feed (in JSON format) that conforms with [RFC8322]. You MAY include the CSAF feed in a Service Document and create a Category Document that further dissects CSAF documents by their document categories.

An example of a CSAF ROLIE feed:

{
  "feed": {
    "id": "example-csaf-feed",
    "title": "Example CSAF feed",
    "link": [
      {
        "rel": "self",
        "href": "https://psirt.domain.tld/advisories/csaf/feed.json"
      }
    ],
    "category": {
      "scheme": "urn:ietf:params:rolie:category:information-type",
      "term": "security advisory"
    },
    "updated": "2021-01-01T12:00Z",
    "entry": [
      {
        "id": "2020-ESA-001",
        "title": "Example Security Advisory 001",
        "link": [
          {
            "rel": "self",
            "href": "https://psirt.domain.tld/advisories/csaf/2020/ESA-001.json"
          }
        ],
        "published": "2021-01-01T11:00Z",
        "updated": "2021-01-01T12:00Z",
        "summary": {
          "content": "Vulnerabilities fixed in ABC 0.0.1"
        },
        "content": {
          "type": "application/json",
          "src": "https://psirt.domain.tld/advisories/csaf/2020/ESA-001.json"
        },
        "format": {
          "schema": "https://raw.githubusercontent.com/oasis-tcs/csaf/master/csaf_2.0/json_schema/csaf_json_schema.json",
          "version": "2.0"
        }
      }
    ]
  }
}
```

Reply:
Is there a specific reason why we want to skip the service document?

I have a bunch of use cases in mind where that would help (e.g. the TLP:AMBER vs. TLP:WHITE discussion)... From my understanding it doesn't really hurt ;-) But I might be wrong here.

Reply:
To me, a service document would be created across an entire set of security content where each type (e.g. CSAF documents, OVAL documents, CVE documents, SBOM documents, etc.) is represented as a separate feed. And for that kind of a service document, you're well outside the scope of CSAF.

Though, I guess you're thinking of a way more granular use case where the service document would advertise several disparate CSAF feeds that would contain subsets of all CSAF documents with certain properties (e.g. TLP), right?

In any case, adding all CSAF ROLIE feeds to a service document certainly is not discouraged, but I wouldn't make it a requirement in this section.

Reply:
I agree that such broad service document would be out of scope for CSAF.

> Though, I guess you're thinking of a way more granular use case where the service document would advertise several disparate CSAF feeds that would contain subsets of all CSAF documents with certain properties (e.g. TLP), right?

Yes. As we require that CSAF documents of different TLP versions must not be present under the same URL they should not be present in the same feed. (Otherwise, you can't access some documents without knowing why or even worse - you're able to access TLP:RED stuff that you should not have access to by accident...) Therefore, I would like to provide the opportunity for different feeds. To me, the RFC sounded like I have to do that via a service document (see [B.1](https://datatracker.ietf.org/doc/html/rfc8322#appendix-B.1)). Is there another way to do that?

> As a side note: I thought we could then use the categories for filtering. E.g. a big manufacturer puts each product family into a single category. Now the end user only selects a subset of categories which are relevant for him. Or one wants to group them by profile type.
> Nevertheless, an issuing party shall provide one feed with all CSAF documents (but based on their TLP level).
> Thoughts?

IMHO, the CSAF field in the security.txt should always link to same type of document to make it easier for automation. If we can provide the standard service document which is used by 80% - awesome.